is/jsonable_test.ts

```diff
@@ -0,0 +1,117 @@
+import { assertEquals } from "@std/assert";
+import { isJsonable } from "./jsonable.ts";
+
+const testcases: [name: string, value: unknown][] = [
+  undefined,
+  null,
+  "",
+  0,
+  true,
+  [],
+  {},
+  0n,
+  () => {},
+  Symbol(),
+].map((x) => {
+  const t = typeof x;
+  switch (t) {
+    case "object":
+      if (x === null) {
+        return ["null", x];
+      } else if (Array.isArray(x)) {
+        return ["array", x];
+      }
+      return ["object", x];
+  }
+  return [t, x];
+});
+
+Deno.test("isJsonable", async (t) => {
+  for (const [name, value] of testcases) {
+    switch (name) {
+      case "undefined":
+      case "null":
+      case "bigint":
+      case "function":
+      case "symbol":
+        await t.step(`return false for ${name}`, () => {
+          assertEquals(isJsonable(value), false);
+        });
+        break;
+      default:
+        await t.step(`return true for ${name}`, () => {
+          assertEquals(isJsonable(value), true);
+        });
+    }
+  }
+
+  for (const [name, value] of testcases) {
+    switch (name) {
+      case "bigint":
+      case "function":
+      case "symbol":
+        await t.step(
+          `return false for ${name} even if it has wrapped by Object.assign`,
+          () => {
+            assertEquals(
+              isJsonable(
+                Object.assign(value as NonNullable<unknown>, { a: 0 }),
+              ),
+              false,
+            );
+          },
+        );
+    }
+  }
+
+  for (const [name, value] of testcases) {
+    switch (name) {
+      case "bigint":
+      case "function":
+      case "symbol":
+        await t.step(
+          `return true for ${name} if it has own toJSON method`,
+          () => {
+            assertEquals(
+              isJsonable(
+                Object.assign(value as NonNullable<unknown>, {
+                  toJSON: () => "custom",
+                }),
+              ),
+              true,
+            );
+          },
+        );
+    }
+  }
+
+  for (const [name, value] of testcases) {
+    switch (name) {
+      case "bigint":
+      case "function":
+      case "symbol":
+        await t.step(
+          `return true for ${name} if it class defines custom toJSON method`,
+          () => {
+            // deno-lint-ignore no-explicit-any
+            (value as any).constructor.prototype.toJSON = () => "custom";
+            try {
+              assertEquals(isJsonable(value), true);
+            } finally {
+              // deno-lint-ignore no-explicit-any
+              delete (value as any).constructor.prototype.toJSON;
+            }
+          },
+        );
+    }
+  }
```

Comment:
[The spec of SerializeJSONProperty](https://tc39.es/ecma262/multipage/structured-data.html#sec-serializejsonproperty) says:
> 2. If value [is an Object](https://tc39.es/ecma262/multipage/ecmascript-data-types-and-values.html#sec-object-type) or value [is a BigInt](https://tc39.es/ecma262/multipage/ecmascript-data-types-and-values.html#sec-ecmascript-language-types-bigint-type), then
>   a. Let toJSON be ? [GetV](https://tc39.es/ecma262/multipage/abstract-operations.html#sec-getv)(value, "toJSON").
>   b. If [IsCallable](https://tc39.es/ecma262/multipage/abstract-operations.html#sec-iscallable)(toJSON) is `true`, then
>     i. Set value to ? [Call](https://tc39.es/ecma262/multipage/abstract-operations.html#sec-call)(toJSON, value, « key »).

So, expected results are:
- bigint: `true`
- function: `true`
- symbol: `false`